internal/servers/destinations.go

```diff
@@ -115,7 +115,11 @@ func (s *DestinationServer) Write2(msg pb.Destination_Write2Server) error {
 			if err := eg.Wait(); err != nil {
 				s.Logger.Error().Err(err).Msg("failed to wait")
 			}
-			return ctx.Err()
+			st := status.FromContextError(ctx.Err())
+			if st == nil {
+				return nil
+			}
+			return st.Err()
```

Comment:
```suggestion
			return status.FromContextError(ctx.Err()).Err()
```
https://github.com/grpc/grpc-go/blob/master/internal/status/status.go#L96-L101

Reply:
Is it guaranteed that that `ctx.Err()` is always not `nil`? Otherwise `st` can be `nil` too

Reply:
See this additional piece: https://github.com/grpc/grpc-go/blob/master/internal/status/status.go#L72-L77
Basically, if we pass a nil err (works either for `status.FromError` and for `status.FromContextError` we'll get `(*status.Status)(nil)`. However, as the receiver for `Code` & `Err` methods may be nil and the code accounts for that, you don't beed to do branching here – you'll just get the nil err on the output and it's what you're doing via `if` statement anyways.

Reply:
OK so I think what I was missing is https://groups.google.com/g/golang-nuts/c/wcrZ3P1zeAk/m/WI88iQgFMvwJ. Thanks for explaining